joomla/joomla-cms
# [4.2] Inject the database into the finder classes

### Summary of Changes
Ongoing effort to inject the database into components and get rid of the deprecated Factory::getDbo(); calls. This pr injects the database in the finder component instead of using the factory to get it from.

### Testing Instructions
- Make sure the smart search content plugin is enabled
- Edit an article and add a description text with the word hello
- Save the article
- Go to the url /administrator/index.php?option=com_finder&view=index and hit the Index button
- Create a search module for the front
- Search for "hello"

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.
